### PR TITLE
Fix Issue 21122 - __traits(getAttributes) wrong scope on enums

### PR DESCRIPTION
**Pull request by BorisCarvajal**
A better approach, previous fix #11163 was not enough.

### CODE REVIEW

File: src/dmd/parse.d [hunk 1]
```diff
@@ -3222,20 +3222,24 @@ final class Parser(AST) : Lexer
...
                 }
 
-                auto em = new AST.EnumMember(loc, ident, value, type, stc, uad, dd);
+                auto em = new AST.EnumMember(loc, ident, value, type, stc, null, dd);
                 e.members.push(em);
 
+                if (udas)
+                {
+                    auto s = new AST.Dsymbols();
+                    s.push(em);
```
Comment (dkorpel) [inline src/dmd/parse.d:3238]: Looks like this introduced a regression (https://github.com/dlang/dmd/pull/14522), was this addition needed for something?

Comment (BorisCarvajal) [reply src/dmd/parse.d:3238]: I don't remember exactly, I think it was just for completeness, the Uda holding the symbol and the symbol holding the uda.
If it works without it, just remove it. `syntaxCopy/arraySyntaxCopy` recursion bugs suck.
